cf/commands/domain/domains_test.go

```diff
@@ -1,137 +1,379 @@
-				[]string{"No domains found"},
-			))
-		})
+			Context("when router api endpoint is not set", func() {
+				BeforeEach(func() {
+					factory.NewRoutingAPIRequirementReturns(
+						&failingRequirement{
+							Name: "RoutingApiRequirement",
+							ui:   ui,
+						},
+					)
+				})
 
-		It("fails when the domains API returns an error", func() {
-			domainRepo.ListDomainsForOrgReturns(errors.New("an-error"))
-			runCommand()
+				It("does not panic", func() {
+					Expect(func() { cmd.Execute(flagContext) }).NotTo(Panic())
 
-			Expect(ui.Outputs).To(ContainSubstrings(
-				[]string{"Getting domains in org", "my-org", "my-user"},
-				[]string{"FAILED"},
-				[]string{"Failed fetching domains"},
-				[]string{"an-error"},
-			))
+				})
+
+				It("prints domain information", func() {
+					cmd.Execute(flagContext)
+					Expect(ui.Outputs).To(ContainSubstrings(
+						[]string{"Private-domain1", "owned"},
+						[]string{"Shared-domain1", "shared"},
+					))
+				})
+
+			})
+
+			Context("when a shared domain with router group is found", func() {
+
+				Context("when routing api endpoint is not set", func() {
+					BeforeEach(func() {
+						factory.NewRoutingAPIRequirementReturns(
+							&failingRequirement{
+								Name: "RoutingApiRequirement",
+								ui:   ui,
+							},
+						)
+
+						domainFields = []models.DomainFields{
+							models.DomainFields{Shared: true,
+								Name:            "Shared-domain1",
+								RouterGroupGuid: "router-group-guid"},
+						}
+						domainRepo.ListDomainsForOrgStub = callBackFunc
+						cmd.Requirements(factory, flagContext)
+					})
+
+					It("panics with missing routing api", func() {
+						Expect(func() { cmd.Execute(flagContext) }).To(Panic())
+						Expect(ui.Outputs).To(ContainSubstrings(
+							[]string{"Routing API URI missing."},
+						))
+					})
+
+				})
+
+				Context("when routing api endpoint is set", func() {
+					Context("when router group does not exist", func() {
+						BeforeEach(func() {
+							domainFields = []models.DomainFields{
+								models.DomainFields{Shared: true,
+									Name:            "Shared-domain1",
+									RouterGroupGuid: "router-group-guid"},
+							}
+							domainRepo.ListDomainsForOrgStub = callBackFunc
+							cmd.Requirements(factory, flagContext)
+							Expect(func() { cmd.Execute(flagContext) }).To(Panic())
+						})
+
+						It("prints the invalid router group message", func() {
+							Expect(ui.Outputs).To(ContainSubstrings(
+								[]string{"FAILED"},
+								[]string{"Invalid router group guid: router-group-guid"},
+							))
+						})
+
+						It("does not print table header", func() {
+							Expect(ui.Outputs).ToNot(ContainSubstrings(
+								[]string{"name", "status", "type"},
+							))
+						})
+					})
+
+					Context("when list router groups returns error", func() {
+						BeforeEach(func() {
+							routingApiRepo.ListRouterGroupsReturns(errors.New("router-group-err"))
+							domainFields = []models.DomainFields{
```

Comment:
The setup of domainFields here should happen in a higher-level context. That it's being repeated for every context within the empty context beginning on line 286 is a smell.